source/versioned-api/versioned-api.rst

```diff
@@ -0,0 +1,277 @@
+
+This specification and subsequent specifications will define the known API
+versions. Drivers SHOULD define an enumeration containing the known API
+versions, using the version identifiers given. Drivers MAY deviate from the
+version identifiers used in this and subsequent specifications if doing so is
+necessary given the driver's programming language's constraints.
+
+Drivers for languages that don't have enums (e.g. PHP) MUST expose the version
+as a string, but SHOULD offer constants to allow for IDE features such as code
+completion. In these cases, the driver MUST validate (e.g. when the application
+provides a version string to the ``ServerApi`` class) that the version string is
+valid and trigger a client-side error if an unknown API version was used.
+
+
+ServerApi class
+~~~~~~~~~~~~~~~
+
+The ``ServerApi`` class stores an API version, along with flags that decide
+whether or not unknown or deprecated commands in the specified API version
+trigger a server-side error. A ``version`` MUST be specified when declaring an
+API version, while the ``strict`` and ``deprecationErrors`` options are both
+optional and default to ``false``. The ``ServerApi`` class is considered
+immutable; changes to the declared API version MUST be prohibited.
+
+
+Declared Version Inheritance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Drivers MUST ensure that users cannot override the API version declared in the
+``MongoClient`` instance. This includes the ``MongoDatabase`` and
+``MongoCollection`` classes, as well as any operations in these classes. See the
+rationale for more details.
+
+
+Sending Declared API Version to the Server
+------------------------------------------
+
+The declared API version MUST be sent to the server with every command, with the
+exception of the cases listed below. Drivers MUST NOT use a server's reported
+``maxWireVersion`` to decide whether it supports the versioned API. The server
+will reply with an error if the declared API version is not supported. If the
+user does not declare an API version, the driver MUST NOT send any API
+versioning options to the server.
+
+
+Command Syntax
+~~~~~~~~~~~~~~
+
+The options from the declared API version are mapped to the following command
+options:
+
+===================== ========================
+**ServerApi field**   **Command option**
+``version``           ``apiVersion``
+``strict``            ``apiStrict``
+``deprecationErrors`` ``apiDeprecationErrors``
+===================== ========================
+
+If an API version was declared, drivers MUST add the ``apiVersion`` option to
+every command that is sent to a server. Drivers MUST add the ``apiStrict`` and
+``apiDeprecationErrors`` options if they were specified by the user, even when
+the specified value is equal to the server default. Drivers SHOULD NOT add these
+options if the user did not specify a value.
+
+
+Cursors
+~~~~~~~
+
+The ``getMore`` command does not accept API parameters; cursors inherit their
```

Comment:
Consider an explicit note about `killCursors` accepting API parameters.  It's weird that one does and the other doesn't, so I think it's worth calling out.

Reply:
Added in a new paragraph:
> In contrast, the ``killCursors`` command accepts API parameters and drivers MUST include them as they would with all other commands.